### ISSUE TITLE
Error after upgrading to 2.4.4 for class name starting with underscore

### ISSUE THREAD

**Issue by zfernandez88** [immutables/immutables#581]
Hello, after upgrading to 2.4.4, compiling the following code causes an error during compilation:
```Java
import org.immutables.value.Value;
import java.util.Optional;

@Value.Immutable
public interface ReferencingUnderscore {
    interface _Type {}

    // Both of these fail
    _Type getType();
    Optional<_Type> getTypeOptional();
}
```

Similar to #574 (in the parsing code), but different cause/error. Not a big fan of the leading underscore, but unfortunately slightly outside of my control. 
```
[ERROR] org.immutables.value.internal.$processor$.$Processor threw java.lang.IllegalArgumentException: Cannot parse type from input string 'java.util.Optional<com.zmf.immutables.ReferencingUnderscore._Type>'. expected '>' but got 'Type'
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser$1Reader.expect($Type.java:847)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser$1Reader.arguments($Type.java:800)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser$1Reader.type($Type.java:728)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser.doParse($Type.java:852)
  	at org.immutables.value.internal.$processor$.encode.$Type$Parser.parse($Type.java:700)
  	at org.immutables.value.internal.$processor$.meta.$FromSupertypesModel.isEligibleFromType($FromSupertypesModel.java:136)
  	at org.immutables.value.internal.$processor$.meta.$FromSupertypesModel.<init>($FromSupertypesModel.java:82)
  	at org.immutables.value.internal.$processor$.meta.$ValueType.getBuildFromTypes($ValueType.java:1228)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables._t18__generateBuilder($Generator_Immutables.java:4812)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables$FragmentDispatch.run($Generator_Immutables.java:16115)
  	at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
  	at org.immutables.value.internal.$generator$.$Intrinsics.$($Intrinsics.java:96)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables._t4__generateImmutable($Generator_Immutables.java:579)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables$FragmentDispatch.run($Generator_Immutables.java:16101)
  	at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
  	at org.immutables.value.internal.$generator$.$Intrinsics.$($Intrinsics.java:96)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables$1.run($Generator_Immutables.java:23)
  	at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
  	at org.immutables.value.internal.$generator$.$Output$7.invoke($Output.java:163)
  	at org.immutables.value.internal.$generator$.$Intrinsics.$($Intrinsics.java:96)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables._t0__generate($Generator_Immutables.java:20)
  	at org.immutables.value.internal.$processor$.$Generator_Immutables$FragmentDispatch.run($Generator_Immutables.java:16097)
  	at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
  	at org.immutables.value.internal.$generator$.$AbstractGenerator.invoke($AbstractGenerator.java:57)
  	at org.immutables.value.internal.$processor$.$Processor.process($Processor.java:63)
  	at org.immutables.value.internal.$generator$.$AbstractGenerator.process($AbstractGenerator.java:87)
  	at org.immutables.processor.ProxyProcessor.process(ProxyProcessor.java:72)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
```

**Comment by elucash** [comment]
should work now with 2.4.6, please give it a try

**Comment by zfernandez88** [comment]
Looks fixed in 2.4.6, thank you!
